Fix SVG currentcolor resolution

mdubet opened the pull request:
#### 92afb29672fbc265024897612b6b4e944a811f84
<pre>
Fix SVG currentcolor resolution
<a href="https://bugs.webkit.org/show_bug.cgi?id=227219">https://bugs.webkit.org/show_bug.cgi?id=227219</a>

Reviewed by Sam Weinig and Darin Adler.

The currentcolor value for fill/stroke properties
has to be resolved at use-time,
not at builder (setter) time.

Remove duplicated color-prop-05-t test.

* LayoutTests/TestExpectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/color-prop-05-t-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fallback-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fill-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-stroke-expected.txt:
* LayoutTests/platform/gtk/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.png: Removed.
* LayoutTests/platform/gtk/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/import/color-prop-05-t-manual-expected.txt:
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fallback-expected.txt:
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fill-expected.txt:
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-stroke-expected.txt:
* LayoutTests/platform/ios/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/mac-monterey-wk2-lbse-text/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/mac-monterey-wk2-pixel/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.png: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/color-prop-05-t-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fallback-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fill-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-stroke-expected.txt:
* LayoutTests/platform/mac/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.png: Removed.
* LayoutTests/platform/mac/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/win/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/svg/W3C-SVG-1.1-SE/color-prop-05-t.svg: Removed.
* LayoutTests/svg/W3C-SVG-1.1/color-prop-05-t-w3c.png: Removed.
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::colorAndSVGPaintType):

Canonical link: <a href="https://commits.webkit.org/256447@main">https://commits.webkit.org/256447@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fdc01f7cc3a74250e1fae199c4fb8ab0de1ff28d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95779 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5031 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28820 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105357 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165664 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99762 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5114 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33807 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88163 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101187 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101440 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3760 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82390 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30819 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85620 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73650 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39523 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19068 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37213 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20390 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4457 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41274 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43319 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39642 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

weinig reviewed (approved):
Nice. Always fun when a comment tells you how to fix a bug :).

mdubet committed:
Fix SVG currentcolor resolution
https://bugs.webkit.org/show_bug.cgi?id=227219

Reviewed by Sam Weinig and Darin Adler.

The currentcolor value for fill/stroke properties
has to be resolved at use-time,
not at builder (setter) time.

Remove duplicated color-prop-05-t test.

* LayoutTests/TestExpectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/color-prop-05-t-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fallback-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fill-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-stroke-expected.txt:
* LayoutTests/platform/gtk/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.png: Removed.
* LayoutTests/platform/gtk/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/import/color-prop-05-t-manual-expected.txt:
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fallback-expected.txt:
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fill-expected.txt:
* LayoutTests/platform/ios-simulator/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-stroke-expected.txt:
* LayoutTests/platform/ios/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/mac-monterey-wk2-lbse-text/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/mac-monterey-wk2-pixel/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.png: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/color-prop-05-t-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fallback-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-fill-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/painting/currentColor-override-pserver-stroke-expected.txt:
* LayoutTests/platform/mac/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.png: Removed.
* LayoutTests/platform/mac/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/win/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/svg/W3C-SVG-1.1-SE/color-prop-05-t-expected.txt: Removed.
* LayoutTests/svg/W3C-SVG-1.1-SE/color-prop-05-t.svg: Removed.
* LayoutTests/svg/W3C-SVG-1.1/color-prop-05-t-w3c.png: Removed.
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::colorAndSVGPaintType):

Canonical link: https://commits.webkit.org/256447@main
